x-pack/elastic-agent/pkg/agent/application/info/agent_metadata.go

```diff
@@ -0,0 +1,115 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package info
+
+import (
+	"fmt"
+	"os"
+	"runtime"
+	"strings"
+
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/release"
+	"github.com/elastic/go-sysinfo"
+	"github.com/elastic/go-sysinfo/types"
+)
+
+// List of variables available to be used in constraint definitions.
+const (
+	// `agent.id` is a generated (in standalone) or assigned (in fleet) agent identifier.
+	agentIDKey = "agent.id"
+	// `agent.version` specifies current version of an agent.
+	agentVersionKey = "agent.version"
+
+	// `os.family` defines a family of underlying operating system (e.g. redhat, debian, freebsd, windows).
+	osFamilyKey = "os.family"
+	// `os.kernel` specifies current version of a kernel in a semver format.
+	osKernelKey = "os.kernel"
+	// `os.platform` specifies platform agent is running on (e.g. centos, ubuntu, windows).
+	osPlatformKey = "os.platform"
+	// `os.version` specifies version of underlying operating system (e.g. 10.12.6).
+	osVersionKey = "os.version"
+	// `os.name` is a operating system name.
+	// Currently we just normalize the name (i.e. macOS, Windows, Linux). See https://www.elastic.co/guide/en/ecs/current/ecs-os.html
+	osNameKey = "os.name"
+	// `os.full` is an operating system name, including the version or code name.
+	osFullKey = "os.full"
+
+	// `host.architecture` defines architecture of a host (e.g. x86_64, arm, ppc, mips).
+	hostArchKey = "host.architecture"
+	// `host.hostname` specifies hostname of the host.
+	hostHostnameKey = "host.hostname"
+	// `host.name` specifies hostname of the host.
+	hostNameKey = "host.name"
+	// `host.id` is a Unique host id.
+	// As hostname is not always unique, use values that are meaningful in your environment.
+	hostIDKey = "host.id"
+	// `host.ip` is Host ip addresses.
+	// Note: this field should contain an array of values.
+	hostIPKey = "host.ip"
+	// `host.mac` is Host mac addresses.
+	// Note: this field should contain an array of values.
+	hostMACKey = "host.mac"
+)
+
+// ECSMetadata returns an agent ECS compliant metadata.
+func (i *AgentInfo) ECSMetadata() (map[string]interface{}, error) {
+	hostname, err := os.Hostname()
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: remove these values when kibana migrates to ECS
```

Comment:
Is there an issue we can link to?

Reply:
@nchaulet can i remove those and replace with ECS compliant variants?

Reply:
agent hostname/name is missing on ingest management page so it needs to stay here for now

Reply:
linked issue https://github.com/elastic/kibana/issues/64173